test/valid.jl

```diff
@@ -58,3 +58,28 @@ end
 @check_concrete struct V52{T1}
     x::Union{T1, Int, Float64}
 end
+
+@check_concrete mutable struct V53{T1}
+    const x::T1
+    y::Vector{Any}
+end
+
+@check_concrete Base.@kwdef mutable struct V54{T1}
+    const x::T1
+    y::Vector{Any} = []
+    const z::Int = 3
+end
+
+@test V54{Int}(; x = 2) isa V54
+@test isconst(V54, :x)
+
+"Struct definition."
+Base.@kwdef @check_concrete mutable struct V55{T1}
+    const x::T1
+    y::Vector{Any} = []
+    const z::Int = 3
+end
+
+@test V55{Int}(; x = 2) isa V55
+@test isconst(V55, :x)
+@test "Struct definition.\n" == string(@eval @doc V55)
```

Comment:
This guy seems to fail Julia 1.0, perhaps we can remove the test.
By design, the macro should propagate struct docstrings anyways.

Reply:
Otherwise all tests pass.

Reply:
The test suite with "Julia 1" actually corresponds to 1.11 (the latest stable version), not 1.0. So it's important to get it to pass.

Reply:
Oh yeah, I checked and forgot to comment again. It's caused by running with `pkg> test` which differs from the standard evaluation.

But the docstrings are well propagated, just that we need another test to avoid a failure (or remove it).

Reply:
Out of curiosity, do you know why running this inside Pkg.test fails?
I think we can skip that test if its failure is not informative.

Reply:
It's actually because it is run in non-interactive mode, and `Core.@doc` gives a `Base.Docs.DocStr` in this case, while it gives a `Markdown.MD` in interactive mode.
I'll remove it.